From 506c1c6296719c3d4afe462a341c59e3f11280ff Mon Sep 17 00:00:00 2001
From: Anilkumar Kolli <akolli@codeaurora.org>
Date: Fri, 25 Sep 2020 19:29:19 +0530
Subject: [PATCH 2/2] ath11k: qcn9000 uses memory with in 32bit addr space

qcn9000 uses memory with in 32 bit address space.
set dma mask to 32 to allocate memory with in 32-bit.

Signed-off-by: Anilkumar Kolli <akolli@codeaurora.org>
---
 drivers/net/wireless/ath/ath11k/pci.c | 2 +-
 drivers/net/wireless/ath/ath11k/pci.h | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath11k/pci.c b/drivers/net/wireless/ath/ath11k/pci.c
index bc5d0566ed52..a6c8fdca8e5b 100644
--- a/drivers/net/wireless/ath/ath11k/pci.c
+++ b/drivers/net/wireless/ath/ath11k/pci.c
@@ -956,7 +956,7 @@ static void ath11k_pci_disable_msi(struct ath11k_pci *ar_pci)
 
 static int ath11k_pci_claim(struct ath11k_pci *ar_pci, struct pci_dev *pdev)
 {
-	u32 pci_dma_mask = PCI_DMA_MASK_64_BIT;
+	u32 pci_dma_mask = PCI_DMA_MASK_32_BIT;
 	struct ath11k_base *ab = ar_pci->ab;
 	u16 control;
 	u16 device_id;
diff --git a/drivers/net/wireless/ath/ath11k/pci.h b/drivers/net/wireless/ath/ath11k/pci.h
index 693f85c951e4..184c83e9d3f4 100644
--- a/drivers/net/wireless/ath/ath11k/pci.h
+++ b/drivers/net/wireless/ath/ath11k/pci.h
@@ -14,7 +14,6 @@
 #define QCN90XX_VENDOR_ID		0x17CB
 #define QCN90XX_DEVICE_ID		0x1104
 #define PCI_BAR_NUM			0
-#define PCI_DMA_MASK_64_BIT		64
 #define PCI_DMA_MASK_32_BIT		32
 
 #define MAX_UNWINDOWED_ADDRESS 0x80000
-- 
2.7.4

